lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1535,6 +1542,18 @@ ChangeResult TypeAnalyzer::visitConstantTensorAllocLikeOp(
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+template <typename OpTy>
+ChangeResult TypeAnalyzer::visitConstantTensorNewLikeOp(
+    OpTy op, ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto input = operands[0]->getValue();
+  auto knowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  knowledge.hasSizes = true;
```

Comment:
this should be removed. `knowledge.hasSizes` is set in `  fillInSizesGivenSizesList`